### ISSUE TITLE
Intermittent ENOENT errors

### ISSUE THREAD

**Issue by ravishivt** [Unitech/PM2#1495]
I am getting a ENOENT error every time pm2 uses the first process instance in a cluster.  For example, if I have 4 instances configured, refreshing the webpage will always give an ENOENT error on my first and fifth reload but are successful the other three times.  Likewise, if I configure for 5 instances, it will fail on the first and sixth reload.

The full stack trace is:

```
Error: ENOENT, no such file or directory
   at Error (native)
   at process.cwd (/var/esupport/es_trunk/node_modules/express-handlebars/node_modules/graceful-fs/polyfills.js:8:19)
   at process.cwd (/var/esupport/es_trunk/node_modules/less-middleware/node_modules/less/node_modules/graceful-fs/polyfills.js:8:19)
   at Object.posix.resolve (path.js:435:50)
   at ExpressHandlebars._getDir (/var/esupport/es_trunk/node_modules/express-handlebars/lib/express-handlebars.js:253:20)
   at ExpressHandlebars.getTemplates (/var/esupport/es_trunk/node_modules/express-handlebars/lib/express-handlebars.js:137:17)
   at ExpressHandlebars.<anonymous> (/var/esupport/es_trunk/node_modules/express-handlebars/lib/express-handlebars.js:76:22)
   at Array.map (native)
   at ExpressHandlebars.getPartials (/var/esupport/es_trunk/node_modules/express-handlebars/lib/express-handlebars.js:54:33)
   at ExpressHandlebars.renderView (/var/esupport/es_trunk/node_modules/express-handlebars/lib/express-handlebars.js:200:14)
   at View.render (/var/esupport/es_trunk/node_modules/express/lib/view.js:93:8)
   at EventEmitter.app.render (/var/esupport/es_trunk/node_modules/express/lib/application.js:566:10)
   at ServerResponse.res.render (/var/esupport/es_trunk/node_modules/express/lib/response.js:938:7)
   at /var/esupport/es_trunk/lib/routes/index.js:10:9
   at Layer.handle [as handle_request] (/var/esupport/es_trunk/node_modules/express/lib/router/layer.js:82:5)
   at next (/var/esupport/es_trunk/node_modules/express/lib/router/route.js:110:13)
   at Route.dispatch (/var/esupport/es_trunk/node_modules/express/lib/router/route.js:91:3)
   at Layer.handle [as handle_request] (/var/esupport/es_trunk/node_modules/express/lib/router/layer.js:82:5)
   at /var/esupport/es_trunk/node_modules/express/lib/router/index.js:267:22
   at Function.proto.process_params (/var/esupport/es_trunk/node_modules/express/lib/router/index.js:321:12)
   at next (/var/esupport/es_trunk/node_modules/express/lib/router/index.js:261:10)
   at Function.proto.handle (/var/esupport/es_trunk/node_modules/express/lib/router/index.js:166:3)
```

My app is defined as follows:

```
{
  "apps" : [{
    "name"        : "ASP",
    "script"      : "index.js",
    "args"        : [],
    "watch"       : true,
    "merge_logs"  : true,
    "instances"   : 4,
    "exec_mode"   : "cluster_mode",
    "max_memory_restart" : "4096M",
    "out_file"    : "/var/log/es/pm2_asp_out.log",
    "error_file"  : "/var/log/es/pm2_asp_err.log",
    "cwd"         : "/var/esupport/es_trunk",
    "env": {
      "NODE_ENV": "PROD"
    }
  }]
}
```

The app successfully launches:

```
[PM2] Process launched
┌──────────┬────┬─────────┬───────┬────────┬─────────┬────────┬─────────────┬──────────┐
│ App name │ id │ mode    │ pid   │ status │ restart │ uptime │ memory      │ watching │
├──────────┼────┼─────────┼───────┼────────┼─────────┼────────┼─────────────┼──────────┤
│ ASP      │ 0  │ cluster │ 24572 │ online │ 0       │ 0s     │ 63.680 MB   │  enabled │
│ ASP      │ 1  │ cluster │ 24577 │ online │ 0       │ 0s     │ 42.625 MB   │  enabled │
│ ASP      │ 2  │ cluster │ 24592 │ online │ 0       │ 0s     │ 39.980 MB   │  enabled │
│ ASP      │ 3  │ cluster │ 24607 │ online │ 0       │ 0s     │ 39.367 MB   │  enabled │
└──────────┴────┴─────────┴───────┴────────┴─────────┴────────┴─────────────┴──────────┘
```

**Comment by Tjatse** [comment]
It seems a problem with express-handler, I just tested it on pure `express` + `less-middleware` and everything goes fine. whats your `ExpressHandlebars ` `[config]`? e.g.:
```
var express = require('express');
var exphbs  = require('express-handlebars');

var app = express();

app.engine('handlebars', exphbs([config]));
```

**Comment by Tjatse** [comment]
try to configure it manually, such as: 
```
app.engine('handlebars', exphbs({
  partialsDir: path.join(__dirname, 'views/partials/')
}));
```

Now the `partialsDir` is an absolute path, but not a relative one (it's `'views/partials/'` by default).

**Comment by ravishivt** [comment]
Thanks for the suggestions.  They didn't seem to make a difference.  But I believe I found the cause of the issue.  I tried to start my app without pm2 and it errored out saying that the port was already in use.  Apparently, I had a standalone server process still running.  My guess is that one process didn't have the specific view at the time it was initialized so it was constantly throwing back ENOENT errors.  Then pm2 started additional processes that didn't have that issue so my issue ended up being intermittent.  Once I stopped that process and restarted pm2, everything worked fine.
